src/cpu/gemm_convolution_utils.cpp

```diff
@@ -1153,6 +1154,11 @@ status_t init_conf(conv_gemm_conf_t &jcp,
                                 : utils::one_of(dst_d.data_type(), s8, u8))
             && weights_d.data_type() == s8;
 
+#if DNNL_AARCH64
+    // For aarch64, int8 conv (signed input data) is not supported.
```

Comment:
Just wondering, if src=s8 isn't supported, shouldn't those cases be left out of the maps in `cpu_convolution_list.cpp`?

Incidentally, the s8s8s8 case is supported in the case of builds with Compute Library - https://github.com/oneapi-src/oneDNN/blob/55ed200a09d412f0e38aed00e23936ccfbb45b93/src/cpu/cpu_convolution_list.cpp#L252

Reply:
Dear nSircombe,

`cpu_convolution_list.cpp`  you refer to is not the latest commit. Would you refer to the latest commit?
https://github.com/fujitsu/oneDNN/blob/7037fbb95a17addf4f9b9d1b803c042f2613e2c7/src/cpu/cpu_convolution_list.cpp

You can check that src=s8 is supported in the latest `cpu_convolution_list.cpp` .

Reply:
Hi, @nSircombe 

Thank you for the comment. We'll check it again. It would be correct to fix it as follows.

```
#if DNNL_AARCH64 && !defined(DNNL_AARCH64_USE_ACL)
```

Reply:
Dear @nSircombe ,

ACL<s8,s8,s8> is not supported now as you mentioned. But, some test codes call gemm_convolution_utils.cpp, and gemm<s8,s8,s8,s32> is run. In order to avoid an error, therefore, we design convolution primitive to execute the reference code of gemm<s8,s8,s8,s32> when gemm<s8,s8,s8,s32> is called.

Reply:
Hi,

> ACL<s8,s8,s8> is not supported now as you mentioned.

Sorry, I don't follow - ACL does provide support for s8s8s8. I'm wondering if these changes, on an ACL enabled build, will cause it to call the ref. impl. in cases which the ACL primitive could have picked up?
> cpu_convolution_list.cpp you refer to is not the latest. Would you refer to the latest commit?

Of course. But my query remains. If `jit_sve_512_x8s8s32x_convolution_fwd_t` doesn't support an s8 source, should it appear in the list above the ACL impl., which does - https://github.com/fujitsu/oneDNN/blob/7037fbb95a17addf4f9b9d1b803c042f2613e2c7/src/cpu/cpu_convolution_list.cpp#L255

> It would be correct to fix it as follows.
> #if !defined(DNNL_AARCH64_USE_ACL) ...

I think so.

Reply:
Hi, @nSircombe 

We are working on the code to fix it correctly. Please wait a little longer.

Reply:
Super, thanks for the update @kawakami-k.

Reply:
Dear @nSircombe ,

We have updated the code for ACL enabled build.

Reply:
Hi, @nSircombe 
We have force updated this PR.

The check code
https://github.com/oneapi-src/oneDNN/commit/7c08aa43dbb238d0610a7c3641e7f038dd2bda78#diff-8fcc2c0ed19739178b74a5f83563f9d506b9ec1055844658c345489f0115f4e9R1157 has been removed, as it's not necessary. Instead, we have added https://github.com/oneapi-src/oneDNN/pull/934/commits/27d8dcbae91f6d7898a32aa64135aad473675b79 , because JIT-ed code for AArch64 doesn't use the compensation for input of (s8 src and s8 weight) and use the compensation for input of (u8 src and s8 weight). The oneDNN hehaviour should be the same as the before for the ACL and other gemm libraries. 

Is this PR godd with you?

Reply:
Hi @kawakami-k, @t-hasimo,

This looks ok to me now, thanks for the updates & explanation.

...just summoning @diaena in case I'm missing something.

Reply:
Looks good to me, thanks!